src/ipc/ipc-common/IPCProxy.h

```diff
@@ -64,15 +64,21 @@ class IPCProxy : public WrapperType, public IPCProxyNewBase
                 if (m_localProviderBinder.provider() == nullptr) {
                     auto serviceName = ipc()->serviceName();
                     auto objPath = ipc()->objectPath();
+                    bool serviceFound = false;
                     for (auto& proxy : m_ipcProxies) {
                         auto proxyAdapterIPCBinder = proxy.ipcBinder;
                         if (proxyAdapterIPCBinder != nullptr) {
+                            proxyAdapterIPCBinder->setEnabled(false); // prevent connecting to server in next two steps
                             proxyAdapterIPCBinder->setObjectPath(objPath);
                             if(!serviceName.isEmpty()) {
                                 proxyAdapterIPCBinder->setServiceName(serviceName);
                             }
+                            proxyAdapterIPCBinder->setEnabled(!serviceFound);
                             proxyAdapterIPCBinder->connectToServer();
```

Comment:
calling "connectToServer()" on a proxy which previously had a "setEnabled(false)" is very weird.

Reply:
Agree. However it is already called twice at this place. Anyway, I will add 'if'.